package-installer/src/main/java/com/vmware/pscoe/iac/installer/Installer.java

```
@@ -1380,68 +1385,124 @@ private static List<Package> getFilesystemPackages(final PackageType type) {
-					"Unable to perform file operation: " + e.getClass().getName() + " : " + e.getLocalizedMessage(), e);
-		} catch (JsonSyntaxException e) {
-			throw new RuntimeException("Unable to parse input / output file data: " + e.getClass().getName() + " : "
-					+ e.getLocalizedMessage(), e);
-		} catch (WorkflowExecutionException e) {
-			try {
-				Files.write(Paths.get(wfErrFilePath), e.getMessage().getBytes(), StandardOpenOption.CREATE);
-			} catch (IOException e1) {
-				throw new RuntimeException(
-						"Unable to perform file operation: " + e.getClass().getName() + " : " + e.getLocalizedMessage(),
-						e);
-			}
-			throw new RuntimeException(
-					"Workflow execution failed: " + e.getClass().getName() + " : " + e.getLocalizedMessage(), e);
-		} catch (Exception e) {
-			throw new RuntimeException("General error during workflow execution: " + e.getClass().getName() + " : "
-					+ e.getLocalizedMessage(), e);
-		}
-	}
+    private static void runWorkflow(final Input input) throws ConfigurationException {
+        input.getText().getTextTerminal().println("Executing vRealize Orchestrator Workflow ...");
+
+        Gson gson = new GsonBuilder().setPrettyPrinting().serializeNulls().create();
+        String wfID = input.get(Option.VRO_RUN_WORKFLOW_ID);
+        String wfInputFilePath = input.get(Option.VRO_RUN_WORKFLOW_INPUT_FILE_PATH);
+        String wfOutputFilePath = input.get(Option.VRO_RUN_WORKFLOW_OUTPUT_FILE);
+        String wfErrFilePath = input.get(Option.VRO_RUN_WORKFLOW_ERR_FILE);
+        wfErrFilePath = StringUtils.isEmpty(wfErrFilePath) && !StringUtils.isEmpty(wfOutputFilePath)
+                ? wfOutputFilePath + ".err"
+                : "workflow.err";
+        String wfTimeout = input.get(Option.VRO_RUN_WORKFLOW_TIMEOUT);
+
+        Properties wfInput = new Properties();
+        try {
+            String wfInputString = new String(Files.readAllBytes(Paths.get(wfInputFilePath)));
+            if (Installer.isValidYaml(wfInputString)) {
```

Idk about this. It's better to check the extension for `.yaml` or `.yml`. That way you won't silently fail

---

Actually my check is better, as the file extension itself is not enough, you can rename a JSON file with YAML extension and your suggestion would fail.